sdk/servicebus/service-bus/src/receivers/receiver.ts

```diff
@@ -421,33 +449,30 @@ export class ReceiverImpl<ReceivedMessageT extends ReceivedMessage | ReceivedMes
 
     return {
       close: async (): Promise<void> => {
-        return this._context.streamingReceiver?.stopReceivingMessages();
+        return this._streamingReceiver?.stopReceivingMessages();
       }
     };
   }
 
   async close(): Promise<void> {
     try {
       this._isClosed = true;
-      if (this._context.namespace.connection && this._context.namespace.connection.isOpen()) {
+      if (this._context.connection && this._context.connection.isOpen()) {
         // Close the streaming receiver.
-        if (this._context.streamingReceiver) {
-          await this._context.streamingReceiver.close();
+        if (this._streamingReceiver) {
+          await this._streamingReceiver.close();
```

Comment:
Cleanup from _context.(streaming|batching)Receivers?

Reply:
The `MessageReceiver.close()` calls a helper method called `_deleteFromCache()` that removes the receiver from the context cache